crossplane/objects.py

```diff
@@ -157,25 +220,25 @@ def __contains__(self, item):
     def get(self, directive):
         return self.index[directive] if directive in self.index else []
 
-    def to_dict(self):
-        result = {}
+    def to_crossplane(self, include_meta=True):
+        result = super(NginxConfigFile, self).to_crossplane(include_meta=include_meta)
 
-        for slot in self.__slots__:
-            if slot not in ('parent', 'index', 'parsed'):
-                result[slot] = getattr(self, slot)
-
-        result['parsed'] = [directive.to_dict() for directive in self.parsed]
+        result['parsed'] = [
+            directive.to_crossplane(include_meta=include_meta)
+            for directive in self.parsed
+        ]
 
         return result
 
 
-class CrossplaneConfig(object):
+class CrossplaneConfig(CrossplaneObject):
     __slots__ = ('index', 'files', 'configs')
+    __meta_slots__ = ('index', 'files')
 
-    def __init__(self, configs=[]):
+    def __init__(self, configs=None):
         self.index = {}
         self.files = []
-        self.configs = configs
+        self.configs = configs if configs is not None else []
```

Comment:
`self.configs` should be `self.config`. I know it doesn't make much sense, but if these objects are supposed to be 1:1 representations of the dicts, we should keep their keys consistent with the variables.